reference_designs/ipu-xrt/matrix_multiplication_column/aie2.py

```diff
@@ -0,0 +1,246 @@
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2023 AMD Inc.
+
+import aie
+from aie.ir import *
+from aie.dialects.func import *
+from aie.dialects.scf import *
+from aie.dialects.aie import *
+from aie.dialects.aiex import *
+from aie.dialects.extras import memref, arith
+from aie.util import mlir_mod_ctx
+
+
+def my_matmul():
+    M = 256
+    K = 128
+    N = 128
+    m = 64
+    k = 32
+    n = 64
+    r = 4
+    s = 4
+    t = 4
+    word_size_in = 2
+    word_size_out = 2
+
+    n_cores = 4
+
+    A_sz_in_i32s = M * K * word_size_in // 4
+    B_sz_in_i32s = K * N * word_size_in // 4
+    C_sz_in_bytes = M * N * word_size_out
+    C_sz_in_i32s = C_sz_in_bytes // 4
+
+    M_div_m = M // m
+    M_div_m_div_n_cores = M // (m * n_cores)
+    K_div_k = K // k
+    N_div_n = N // n
+    tiles = M_div_m * N_div_n // n_cores
+
+    # Matrix A: MxK, submatrices a: mxk
+    k_in_i32s = k * word_size_in // 4
+    K_in_i32s = K * word_size_in // 4
+    m_x_n_cores = m * n_cores
+
+    # Matrix B: KxN, submatrices b: kxn
+    n_in_i32s = n * word_size_in // 4
+    N_in_i32s = N * word_size_in // 4
+    k_x_N_in_i32s = k * N * word_size_in // 4
+    N_div_n_div_n_cores  = N // n // n_cores
+    n_x_n_cores_in_i32s  = n * n_cores * word_size_in // 4
+
+    # Output Matrix C: MxN
+    n_in_i32s_out = n * word_size_out // 4
+    N_in_i32s_out = N * word_size_out // 4
+    m_x_n_cores_x_N_in_i32s_out = m * n_cores * N * word_size_out // 4
```

Comment:
These variables were necessary in the templating approach, but maybe less so in the Python script, since some of the variable names are almost as long as the expression itself... I'm also not sure if they are all still used

Reply:
A few of these were indeed not used so I removed them. I didn't change the variable names.